app/utils/scheduled_post/scheduled_post.test.ts

```diff
@@ -0,0 +1,140 @@
+    };
+
+    beforeEach(() => {
+        jest.clearAllMocks();
+    });
+
+    it('shows confirmation dialog with correct text and buttons', () => {
+        deleteScheduledPostConfirmation(baseProps);
+
+        const alertCall = jest.mocked(Alert.alert).mock.calls[0];
+        expect(alertCall[0]).toBe('Delete scheduled post');
+        expect(alertCall[1]).toBe('Are you sure you want to delete this scheduled post?');
+
+        // Ensure buttons exist
+        expect(alertCall[2]).toBeDefined();
+        const buttons = alertCall[2] ?? [];
+
+        expect(buttons).toHaveLength(2);
+
+        // Check Cancel button
+        expect(buttons[0]?.text).toBe('Cancel');
+        expect(typeof buttons[0]?.onPress).toBe('function');
+
+        // Check Delete button
+        expect(buttons[1]?.text).toBe('Delete');
+        expect(typeof buttons[1]?.onPress).toBe('function');
+    });
+
+    it('calls deleteScheduledPost when confirmed', () => {
+        deleteScheduledPostConfirmation(baseProps);
+
+        const alertMock = jest.mocked(Alert.alert);
+        expect(alertMock).toHaveBeenCalled();
+
+        const alertCall = alertMock.mock.calls[0];
+        expect(alertCall).toBeDefined();
+
+        const buttons = alertCall[2] ?? [];
+        expect(buttons).toHaveLength(2);
+
+        const confirmButton = buttons[1];
+        expect(confirmButton).toBeDefined();
+        expect(confirmButton.onPress).toBeDefined();
+
+        confirmButton.onPress?.();
+
+        // Verify deleteScheduledPost is called with the correct arguments
+        expect(deleteScheduledPost).toHaveBeenCalledWith(
+            baseProps.serverUrl,
+            baseProps.scheduledPostId,
+        );
+    });
+
+    it('closes swipeable when cancelled', () => {
+        deleteScheduledPostConfirmation(baseProps);
+
+        const alertMock = jest.mocked(Alert.alert);
+        expect(alertMock).toHaveBeenCalled();
+        const alertCall = alertMock.mock.calls[0];
+        expect(alertCall).toBeDefined();
+        const buttons = alertCall[2] ?? [];
+        expect(buttons).toHaveLength(2);
+
+        const cancelButton = buttons[0];
+        expect(cancelButton).toBeDefined();
+        expect(cancelButton.onPress).toBeDefined();
+
+        cancelButton.onPress?.();
+
+        // Verify swipeable close is called
+        expect(baseProps.swipeable?.current?.close).toHaveBeenCalled();
+    });
+
+    it('handles missing swipeable ref', () => {
+        const propsWithoutSwipeable = {
+            ...baseProps,
+            swipeable: undefined,
+        };
+
+        deleteScheduledPostConfirmation(propsWithoutSwipeable);
+
+        // Get the cancel button callback
+        const alertMock = jest.mocked(Alert.alert);
+        const buttons = alertMock.mock.calls[0]?.[2] ?? []; // Ensure buttons array exists
+        const cancelButton = buttons[0];
+        expect(cancelButton).toBeDefined();
+        const onPress = cancelButton.onPress ?? (() => {});
```

Comment:
This is weird. If the `?? (() => {})` part is just to keep typescript happy, you can use the non-null assertion operator (`!` the exclamation mark) to tell TS that you are sure it is never going to be undefined.

The non-null assertion operator is dangerous and should be used scarcely, but in this scenario should be fine.